rohrpost/sync_consumer.py

```diff
@@ -0,0 +1,68 @@
+import json
+from functools import partial
+
+from asgiref.sync import async_to_sync
+from channels.generic.websocket import WebsocketConsumer
+
+from . import handlers
+from .message import send_error
+from .registry import HANDLERS
+
+assert handlers  # silence qa
+
+REQUIRED_FIELDS = ["type", "id"]
+
+try:
+    DECODE_ERRORS: tuple = (json.JSONDecodeError, TypeError)
+except AttributeError:
+    # Python 3.4 raises a ValueError instead of json.JSONDecodeError
+    DECODE_ERRORS = (ValueError, TypeError)
+
+
+class SyncRohrpostConsumer(WebsocketConsumer):
+    def connect(self):
+        self.accept()
+
+    def disconnect(self, close_code):
+        pass
+
+    def receive(self, text_data):
```

Comment:
This was `handle_rohrpost_message` in v1.  Now it has a dependency on channels.  I'm not sure if we should just add a hard dependency on channels or if we should keep rohrpost as independent as possible.

We could still have `handle_rohrpost_message(consumer, text_data)` in `main.py`.

Reply:
Btw. channels requires python >= 3.5 whereas rohrpost currently requires python >= 3.4.